### PR TITLE
Add setup.py with new entry point.

### PR DESCRIPTION
**Pull request by jbenjos**
Will allow poseidon to be PIP installable and no changing PYTHONPATH

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from setuptools import setup, find_packages
+
+setup(
+    name='cr-posideon',
+    version='v0.4.1alpha',
+    packages=find_packages(exclude=['contrib', 'docs', 'tests']),
+    install_requires=['pika==0.11.0', 'requests==2.18.4', 'schedule==0.4.3'],
+    test_require=['pytest==3.2.1', 'pytest-cov==2.5.1', 'pylint==1.7.2', 'httmock==1.2.6', 'mock==2.0.0'],
+    license='MIT',
```
Comment (d-grossman) [inline setup.py]: license does not match project license

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from setuptools import setup, find_packages
+
+setup(
+    name='cr-posideon',
+    version='v0.4.1alpha',
+    packages=find_packages(exclude=['contrib', 'docs', 'tests']),
+    install_requires=['pika==0.11.0', 'requests==2.18.4', 'schedule==0.4.3'],
```
Comment (d-grossman) [inline setup.py]: where do the packages for documentation building fit in for this?

**Comment by d-grossman** [comment]
Currently we deploy though docker containers; testing, documentation generation, and execution all share a consistent environment.  I'm not sure that pip installations of the project is the direction it should go.  I'll talk with @cglewis to get another perspective.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from setuptools import setup, find_packages
+
```
Comment (d-grossman) [inline setup.py]: How do you force a python3+ only environment

**Comment by cglewis** [comment]
Poseidon is not meant to be a library, but instead is an application.  I generally think of pip packages as being used as libraries, so I'm not sure being able to pip install Poseidon really makes a lot of sense in its current (or foreseeable) form.

**Comment by d-grossman** [comment]
At this time Docker containers are the way that Poseidon is installed.

**Comment by jbenjos** [comment]
If the project was a just a couple of independent scripts I wouldn't bother with setuptools. I think it's worth doing because of the current PYTHONPATH and other path manipulation requirements, as well as opening up the ability to clean up some of the interproject imports. There is a very elaborate module hierarchy under ./poseidon that is being managed manually right now. This will help get a handle on that. This will also let me simplify the Dockerfile a bit. I think applications are okay as pip packages, pip itself for instance is a pip package as well as a ton of other command line tools and stuff. But mostly the point is to remove the dependencies on specialized paths and environment variables.

**Comment by jbenjos** [comment]
This is also to partly address #281

**Comment by d-grossman** [comment]
@jbenjos you need to address the items in review before this can move forward.

**Comment by cglewis** [comment]
@jbenjos any update on progress for the review items?

**Comment by d-grossman** [comment]
There has been no movement on this in 22 days.

**Pull request closed without merging**